app/src/main/java/com/kinkong/database/FBDatabase.java

```diff
@@ -83,7 +82,18 @@ public void getAnswersCount(ValueEventListener valueEventListener) {
         database.getReference("answers/" + nextQuestionNum + "/answers_count").addListenerForSingleValueEvent(valueEventListener);
     }
 
-    public void updateNextQuestion() {
+    public boolean upateNextQuestion(boolean isWinner) {
+        boolean hasNextQuestion = false;
+        if (isWinner && shouldSeenNextQuestion()) {
+            updateNextQuestion();
+            hasNextQuestion = true;
+        } else {
+            updateNextQuestionSession();
```

Comment:
both updateNextQuestion() & updateNextQuestionSession() modify nextQuestionNum and then call getQuestionAt
you could remove these 2 private methods, update nextQuestionNum inside the if/else clause according to each case and after the if/end has finished call getQuestionAt...